test/dt_arith.c

```diff
@@ -2904,6 +3420,39 @@ test_conv_flt_1_hw_conv_from_flt16(void *hw_dst, unsigned char *src_buf, size_t
             *((long double *)hw_dst) = (long double)aligned;
             break;
 #endif
+
+#ifdef H5_HAVE_COMPLEX_NUMBERS
+#ifdef H5_HAVE_C99_COMPLEX_NUMBERS
+        case FLT_COMPLEX:
+            *((H5_float_complex *)hw_dst) = (H5_float_complex)aligned;
```

Comment:
Several places in this file deal with casting values differently depending on whether C99 or MSVC complex number types are supported. While the `H5_CMPLXF` macro below could cover both cases, I wanted to exactly match how each platform would handle casting values to avoid running into possible odd behavior and testing failures. The `H5_CMPLXF` macro has issues with specific special floating-point values that can't be fixed without C11 support.